src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -0,0 +1,585 @@
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n, 1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=1, n, 1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    fuse_loops(parser, code)
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_different_loop_vars(parser):
+    '''
+    Test that loop variables are identicla
+    '''
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do ji=1, n
+                 do jj=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    with pytest.raises(TransformationError) as err:
+        fuse_loops(parser, code)
+    assert "Loop variables must be the same, but are 'jj' and 'ji'" \
+        in str(err.value)
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.xfail(reason="Needs evaluation of constant expressions")
+def test_fuse_correct_bounds(parser):
+    '''
+    Test that loop boundaries must be identical.
+    '''
+    # TODO: This test needs evaluation
+    # of constant expressions in PSyclone
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=2-1, n
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=2, n+1-1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              end subroutine sub'''
+    fuse_loops(parser, code)
+
+
+# ----------------------------------------------------------------------------
+def test_fuse_dimension_change(parser):
+    '''Test that inconsistent use of dimemsions are detected, e.g.:
+    loop1:  a(i,j)
+    loop2:  a(j,i)
+    when at least one operation is a write
+    '''
+
+    # The first example can be merged, since 't' is read-only,
+    # so it doesn't matter that it is accessed differently
+    code = '''subroutine sub()
+              integer :: ji, jj, n
+              integer, dimension(10,10) :: s, t
+              do jj=1, n+1
+                 do ji=1, 10
+                    s(ji, jj)=t(ji, jj)+1
+                 enddo
+              enddo
+              do jj=1, n+1
+                 do ji=1, 10
+                    s(ji, jj)=s(ji, jj) + t(jj, jj) + t(ji, ji)
+                 enddo
+              enddo
+              end subroutine sub'''
+
+    fuse_loops(parser, code)
```

Comment:
Please add a quick check that the resulting code is OK.

Reply:
Done.